File: chroma-league/src/main/java/net/booone/chromaleague/tasks/FetchPlayerListTask.java
package net.booone.chromaleague.tasks;

import net.booone.chromaleague.rest.http.LeagueHttpClients;
import net.booone.chromaleague.rest.http.handlers.PlayerListResponseHandler;
import net.booone.chromaleague.state.RunningState;

public class FetchPlayerListTask implements Runnable {
    public static final String URL = "https://127.0.0.1:2999/liveclientdata/playerlist";
    private final PlayerListResponseHandler playerListResponseHandler = new PlayerListResponseHandler();

    @Override
    public void run() {
        LeagueHttpClients.getNonBlockingResponse(URL, playerListResponseHandler)
                .ifPresent(playerList -> RunningState.getGameState().setPlayerList(playerList));
    }
}

File: src/tasks/FetchPlayerListTask.ts
import {getNonBlockingResponse} from '../rest/http/LeagueHttpClients.js';
import type {Player} from '../rest/playerlist/Player.js';
import {PlayerList} from '../rest/playerlist/PlayerList.js';
import {RunningState} from '../state/RunningState.js';

const URL = 'https://127.0.0.1:2999/liveclientdata/playerlist';

export async function fetchPlayerListTask(): Promise<void> {
    const players = await getNonBlockingResponse<Player[]>(URL);
    if (players) {
        RunningState.getGameState().playerList = new PlayerList(players);
    }
}
